RealmSwift/SwiftUI.swift

```diff
@@ -617,6 +617,292 @@ extension Projection: _ObservedResultsValue { }
+public enum ObservedQueryResultsState {
+    /// Subscription has been added and waiting for data to bootstrap.
+    case pending
+    /// An error has occurred while adding the subscription (client or server side).
+    case error(Error)
+    /// Data has been bootstrapped and query results updated.
+    case completed
+}
+
+// MARK: ObservedQueryResults
+
+/// A property wrapper type that represents the results of a query on a realm resulting from a query subscription.
+///
+/// The results use the realm configuration provided by
+/// the environment value `EnvironmentValues/realmConfiguration`. or the configuration injected on the initializer.
+///
+/// `ObservedQueryResults` is mutable, you can add or remove elements from the results,
+/// if this object are out of the view of the subscription query this will not be included on this results
+///
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+@propertyWrapper public struct ObservedQueryResults<ResultType>: DynamicProperty, BoundCollection where ResultType: _ObservedResultsValue & RealmFetchable & KeypathSortable & Identifiable {
+    internal class Storage: ObservableStorage<Results<ResultType>> {
+        var setupHasRun = false
+        private let subscription: ((Query<ResultType>) -> Query<Bool>)?
+        var queryResults: QueryResults<ResultType>?
+
+        private func didSet() {
+            if setupHasRun {
+                setupValue()
+            }
+        }
+
+        func setupValue() {
+            guard let queryResults = queryResults,
+                  let results = queryResults.results else {
+                return
+            }
+
+            value = results
+            if let sortDescriptor = sortDescriptor {
+                value = value.sorted(byKeyPath: sortDescriptor.keyPath, ascending: sortDescriptor.ascending)
+            }
+
+            let filters = [searchFilter, filter ?? `where`].compactMap { $0 }
+            if !filters.isEmpty {
+                let compoundFilter = NSCompoundPredicate(andPredicateWithSubpredicates: filters)
+                value = value.filter(compoundFilter)
+            }
+        }
+
+        var sortDescriptor: SortDescriptor? {
+            didSet {
+                didSet()
+            }
+        }
+
+        var filter: NSPredicate? {
+            didSet {
+                didSet()
+            }
+        }
+        var `where`: NSPredicate? {
+            didSet {
+                didSet()
+            }
+        }
+        var configuration: Realm.Configuration? {
+            didSet {
+                didSet()
+            }
+        }
+
+        var searchString: String = ""
+        var searchFilter: NSPredicate? {
+            didSet {
+                didSet()
+            }
+        }
+
+        func subscribe() {
+            guard let configuration = configuration else {
+                throwRealmException("")
```

Comment:
These need actual errors

Reply:
I removed all the code and went with the ObservedResults approach